### ISSUE TITLE
Crash when adding and removing favorite

### ISSUE THREAD

**Issue by juliushaertl** [stefan-niedermann/nextcloud-notes#988]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**
The app crashes, additionally there is a rendering bug when the favorite was added, could be related.

**To Reproduce**

In the main notes list, mark a note as favorite and remove it again by tapping the star icon

**Expected behavior**

It just works :wink:

**Screenshots**

**Smartphone (please complete the following information):**
 - Nextcloud Notes-Version (android app): 2.17.1
 - F-Droid or Play Store: 
 - Android-Version: 11
 - Device: Pixel 3a

**Server**
 - Nextcloud version: 
 - Nextcloud Notes version (server app): 

**Stacktrace**

```
App Version: 2.17.1
App Version Code: 2017001
App Flavor: play

Files App Version Code: 30140052

---

OS Version: 4.9.223-g90fdcf28fd1c-ab6846391(6869500)
OS API Level: 30
Device: sargo
Manufacturer: Google
Model (and Product): Pixel 3a (sargo)

---

java.lang.ArrayIndexOutOfBoundsException: length=163; index=-1
	at java.util.ArrayList.get(ArrayList.java:439)
	at it.niedermann.owncloud.notes.main.items.ItemAdapter.getItem(ItemAdapter.java:202)
	at it.niedermann.owncloud.notes.main.MainActivity.onNoteFavoriteClick(MainActivity.java:876)
	at it.niedermann.owncloud.notes.main.items.NoteViewHolder.lambda$bindFavorite$2$NoteViewHolder(NoteViewHolder.java:109)
	at it.niedermann.owncloud.notes.main.items.-$$Lambda$NoteViewHolder$rNxJqnoOB-N2wbk5ji9ov7z7Aec.onClick(Unknown Source:2)
	at android.view.View.performClick(View.java:7448)
	at android.view.View.performClickInternal(View.java:7425)
	at android.view.View.access$3600(View.java:810)
	at android.view.View$PerformClick.run(View.java:28305)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

```

**Comment by juliushaertl** [comment]
Hm I cannot upload the screenshot of the UI issue right now, will create a separate issue for that later then.

**Comment by stefan-niedermann** [comment]
Fix will be released with `v2.17.2` - thank you very much @Isaac-Graham !
